### PR TITLE
Fix lifecycle plugin diagnostics being ignored during executable generation

### LINKED ISSUE #41915: [Bug]: Diagnostics reported in CompilerLifecycleEventContext are ignored
### Description

$Subject.

Consider the following code. 
```java
    @Override
    public void perform(CompilerLifecycleEventContext compilerLifecycleEventContext) {
        DiagnosticInfo diagnosticInfo = new DiagnosticInfo(C2CDiagnosticCodes.DOCKER_FAILED.getCode(),
                "Test Diag", DiagnosticSeverity.WARNING);
        Diagnostic diagnostic = DiagnosticFactory.createDiagnostic(diagnosticInfo, new NullLocation());
        compilerLifecycleEventContext.reportDiagnostic(diagnostic);
        
        }
 ```
 
Ideally, this should show a warning to all packages which uses this compiler plugin. but in this case this diagnostic is ignored. 

### Steps to Reproduce

1. Clone https://github.com/xlight05/module-ballerina-c2c/tree/lifecycle-diag
2. Invoke ./gradlew clean build -x check -x test and copy the generated pack
3. Create a sample ballerina project(http hello world) and invoke <generated pack path> build --cloud=docker

### Affected Version(s)

All

### OS, DB, other environment details and versions

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by ShammiL**
Fixes #41915

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by gayaldassanayake — APPROVED** [review]

**Review by Dilhasha — DISMISSED** [review]

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CreateExecutableTask.java [hunk 1]
```diff
@@ -127,16 +126,9 @@ public void execute(Project project) {
...
-            if (!diagnostics.isEmpty()) {
-                //  TODO: When deprecating the lifecycle compiler plugin, we can remove this check for duplicates
-                //   in JBallerinaBackend diagnostics and the diagnostics added to EmitResult.
-                diagnostics = diagnostics.stream()
-                        .filter(diagnostic -> !jBallerinaBackend.diagnosticResult().diagnostics().contains(diagnostic))
-                        .collect(Collectors.toList());
-                if (!diagnostics.isEmpty()) {
-                    diagnostics.forEach(d -> out.println("\n" + d.toString()));
-                }
+            List<Diagnostic> emitDiagnostics = new ArrayList<>(emitResult.diagnostics().diagnostics());
```
Comment (LakshanWeerasinghe) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CreateExecutableTask.java:129]: Why do we need to create a new list here? Can't we just use the returned collection and iterate over it and print the values?

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CreateExecutableTask.java [hunk 2]
```diff
@@ -127,16 +123,9 @@ public void execute(Project project) {
...
-                //  TODO: When deprecating the lifecycle compiler plugin, we can remove this check for duplicates
-                //   in JBallerinaBackend diagnostics and the diagnostics added to EmitResult.
-                diagnostics = diagnostics.stream()
-                        .filter(diagnostic -> !jBallerinaBackend.diagnosticResult().diagnostics().contains(diagnostic))
-                        .collect(Collectors.toList());
-                if (!diagnostics.isEmpty()) {
-                    diagnostics.forEach(d -> out.println("\n" + d.toString()));
-                }
+            // Print diagnostics found during emit executable
+            if (!emitResult.diagnostics().diagnostics().isEmpty()) {
```
Comment (LakshanWeerasinghe) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CreateExecutableTask.java:127]: We don't need the if condition here

Comment (Dilhasha) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CreateExecutableTask.java:127]: We might have to do a null check for `emitResult.diagnostics()` right?

Comment (ShammiL) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CreateExecutableTask.java:127]: According to the current implementation, `emitResult` would contain an arraylist even if it is empty. In that case, do we need a null check?

**Review by Dilhasha — APPROVED** [review]

**Pull request merged**